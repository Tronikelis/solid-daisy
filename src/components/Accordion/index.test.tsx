import { fireEvent, render, screen } from "@solidjs/testing-library";
import { ComponentProps, createSignal } from "solid-js";
import { beforeEach, describe, expect, it, vi } from "vitest";

import { Accordion, AccordionItem } from ".";

describe("<Accordion />", () => {
    let props: ComponentProps<typeof Accordion>;

    beforeEach(() => {
        props = {
            selected: "a",
            setSelected: vi.fn(),
            children: "Bar",
        };
    });

    it("renders correctly", () => {
        const { container } = render(() => (
            <Accordion>
                <AccordionItem value="a" title="Foo">
                    Bar
                </AccordionItem>
                <AccordionItem value="b" title="Foo">
                    Bar
                </AccordionItem>
            </Accordion>
        ));
        expect(container).toMatchSnapshot();
    });

    it("selecting accordion item works", () => {
        const [selected, setSelected] = createSignal("a");

        render(() => (
            <Accordion {...props} selected={selected()} setSelected={setSelected}>
                <AccordionItem value="a" title="A">
                    a
                </AccordionItem>
                <AccordionItem value="b" title="B">
                    b
                </AccordionItem>
            </Accordion>
        ));

        fireEvent.click(screen.getByText("B"));

        // eslint-disable-next-line solid/reactivity
        expect(selected()).toBe("b");
    });
});
