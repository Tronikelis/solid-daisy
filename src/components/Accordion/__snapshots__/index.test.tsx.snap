// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Accordion /> > renders correctly 1`] = `
<div>
  <div
    class="flex flex-col gap-2"
  >
    <div
      class="collapse collapse-arrow bg-base-200"
    >
      <input
        name="cl-0"
        type="radio"
      />
      <div
        class="collapse-title text-xl font-medium"
      >
        Foo
      </div>
      <div
        class="collapse-content"
      >
        Bar
      </div>
    </div>
    <div
      class="collapse collapse-arrow bg-base-200"
    >
      <input
        name="cl-0"
        type="radio"
      />
      <div
        class="collapse-title text-xl font-medium"
      >
        Foo
      </div>
      <div
        class="collapse-content"
      >
        Bar
      </div>
    </div>
  </div>
</div>
`;
